packages/devextreme/js/events/core/events_engine.js

```diff
@@ -274,6 +275,15 @@ function getHandlersController(element, eventName) {
                     }
                 } else {
                     e.currentTarget = e.delegateTarget || e.target;
+
+                    const isTargetInShadowDOM = Boolean(e.target?.shadowRoot);
```

Comment:
This we already do inside of `getEventTarget` as well

Reply:
[commit](https://github.com/DevExpress/DevExtreme/pull/27189/commits/a61f0524d90019a7baca359bd7062f2d419c1f91)